templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -4,199 +4,228 @@ contents:
+        if nmcli conn up ovs-if-br-ex; then
+          echo "OVS successfully configured"
+          ip a show br-ex
+          configure_driver_options ${iface}
+          exit 0
         fi
-        cat <<EOF >> ${new_conn_file}
-    [ovs-interface]
-    type=internal
-    EOF
-        nmcli c load ${new_conn_file}
-        echo "Loaded new ovs-if-br-ex connection file: ${new_conn_file}"
-      else
-        nmcli c add type ovs-interface slave-type ovs-port conn.interface br-ex master ovs-port-br-ex con-name \
-          ovs-if-br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac}
+        sleep 5
+        counter=$((counter+1))
+      done
+
+      echo "ERROR: Failed to activate ovs-if-br-ex NM connection"
+      # if we made it here networking isnt coming up, revert for debugging
+      set +e
+      nmcli conn down ovs-if-br-ex
+      nmcli conn down ovs-if-phys0
+      nmcli conn up $old_conn
+      exit 1
+    elif [ "$1" == "OpenShiftSDN" ]; then
+      # Revert changes made by /usr/local/bin/configure-ovs.sh.
+      # Remove OVS bridge "br-ex". Use the default NIC for cluster network.
+
+      if nmcli connection show ovs-port-phys0 &> /dev/null; then
+        nmcli c del ovs-port-phys0 
       fi
-    fi
 
-    # wait for DHCP to finish, verify connection is up
-    counter=0
-    while [ $counter -lt 5 ]; do
-      sleep 5
-      # check if connection is active
-      if nmcli --fields GENERAL.STATE conn show ovs-if-br-ex | grep -i "activated"; then
-        echo "OVS successfully configured"
-        ip a show br-ex
-        configure_driver_options ${iface}
-        exit 0
+      if nmcli connection show ovs-if-phys0 &> /dev/null; then
+        nmcli c del ovs-if-phys0
       fi
-      counter=$((counter+1))
-    done
-
-    echo "WARN: OVS did not succesfully activate NM connection. Attempting to bring up connections"
-    counter=0
-    while [ $counter -lt 5 ]; do
-      if nmcli conn up ovs-if-br-ex; then
-        echo "OVS successfully configured"
-        ip a show br-ex
-        configure_driver_options ${iface}
-        exit 0
+
+      if nmcli connection show ovs-port-br-ex &> /dev/null; then
+        nmcli c del ovs-port-br-ex
+      fi
+
+      if nmcli connection show ovs-if-br-ex &> /dev/null; then
+        nmcli c del ovs-if-br-ex
       fi
-      sleep 5
-      counter=$((counter+1))
-    done
-
-    echo "ERROR: Failed to activate ovs-if-br-ex NM connection"
-    # if we made it here networking isnt coming up, revert for debugging
-    set +e
-    nmcli conn down ovs-if-br-ex
-    nmcli conn down ovs-if-phys0
-    nmcli conn up $old_conn
-    exit 1
+
+      if nmcli connection show br-ex &> /dev/null; then
+        nmcli c del br-ex
+      fi
+
+      # remove bridges created by ovn-kubernetes, try to delete br-ex again in case NM fail to talk to ovsdb
+      ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local -- --if-exists del-br br-ex
```

Comment:
@pliurh one thing I just thought of is that, if you are using openshift-sdn and you reboot a node, this will run again and delete these bridges. Does openshift-sdn also use br-int bridge? Would this cause problems for openshift-sdn to delete the bridge on each reboot?

@danwinship ^ ?

Reply:
openshift-sdn does not use `br-int`, it only uses `br0`. And it will end up destroying and recreating br0 if OVS brings it up after a reboot anyway. It doesn't want OVS preserving its state across reboots.